APR-Core/QuestHandler.lua

```diff
@@ -3648,91 +3619,47 @@ APR_QH_EventFrame:SetScript("OnEvent", function(self, event, ...)
 		end
 		APR1[APR.Realm][APR.Name]["QuestCounter2"] = APR1[APR.Realm][APR.Name]["QuestCounter2"] + 1
 	end
-	if (event=="UNIT_QUEST_LOG_CHANGED") then
+	if (event == "UNIT_QUEST_LOG_CHANGED") then
 		local arg1, arg2, arg3, arg4, arg5 = ...;
 		if (arg1 == "player" and Updateblock == 0) then
 			Updateblock = 1
 			C_Timer.After(1, APR_UpdQuestThing)
 		end
 	end
-	if (event=="ZONE_CHANGED") then
+	if (event == "ZONE_CHANGED") then
 		QNumberLocal = 0
 		if (APR.ZoneTransfer == 0) then
 			C_Timer.After(2, APR_UpdatezeMapId)
 			C_Timer.After(3, APR_ZoneResetQnumb)
 			APR.BookingList["UpdateMapId"] = 1
 		end
 	end
-	if (event=="ZONE_CHANGED_NEW_AREA") then
+	if (event == "ZONE_CHANGED_NEW_AREA") then
 		if (APR.ZoneTransfer == 0) then
 			C_Timer.After(2, APR_UpdatezeMapId)
 			APR.BookingList["UpdateMapId"] = 1
 		end
 	end
-	if (event=="GOSSIP_CLOSED") then
+	if (event == "GOSSIP_CLOSED") then
 		APRGOSSIPCOUNT = 0
 	end
-	if (event=="QUEST_DETAIL") then -- Fired when the player is given a more detailed view of his quest.
-		local CurStep = APR1[APR.Realm][APR.Name][APR.ActiveMap]
-		local steps
-		if (CurStep and APR.QuestStepList and APR.QuestStepList[APR.ActiveMap]) then
-			steps = APR.QuestStepList[APR.ActiveMap][CurStep]
-		end
-		if (steps and steps["DenyNPC"]) then
-			local npc_id, name = GetTargetID(), UnitName("target")
-			if (npc_id and name) then
-					if (npc_id == steps["DenyNPC"]) then
-						C_GossipInfo.CloseGossip()
-					end
-			end
-		end
-		local target = GetTargetID()
-		if (target) then
-			local CurStep = APR1[APR.Realm][APR.Name][APR.ActiveMap]
-			if (CurStep and APR.ActiveMap and APR.QuestStepList and APR.QuestStepList[APR.ActiveMap] and APR.QuestStepList[APR.ActiveMap][CurStep]) then
-				local steps = APR.QuestStepList[APR.ActiveMap][CurStep]
-				local APRDenied = 0
-				if (steps and steps["DenyNPC"]) then
-					local name = UnitName("target")
-					if (target and name) then
-						if (tonumber(target) == steps["DenyNPC"]) then
-							APRDenied = 1
-						end
-					end
-				end
-				if (APRDenied == 1) then
-					CloseQuest()
-					print("APR: "..L["NOT_YET"])
-				end
-			end
-		end
-		if (GetQuestID() and (APR1[APR.Realm][APR.Name]["Settings"]["AutoAccept"] == 1) and (not IsModifierKeyDown())) then
+	if (event == "QUEST_DETAIL") then -- Fired when the player is given a more detailed view of his quest.		
+		-- Deny NPC
+		CheckDenyNPC(steps)
+		if (GetQuestID()) then

```

Comment:
What is this doing? It just starts an If Then, but does nothing - goes right to the next If Then?

Reply:
I see further down where the CheckDenyNPC(steps) is created, but I do not see anything on this line to indicate what the function is doing.

Reply:
Ok, so the CheckDenyNPC() will chekc if the current targeted NPC is allow to this step or not 
If it's a deny npc 
  we closing the gossip/quest frame +display a message in the chat
else 
  we getting the questID to accept it (or not)

Reply:
Ok, because how that current loop If/Else statement reads like it checks for it, **then** does nothing at that specific line?

Reply:
the function simply adds the piece of code that was there before, it's just that I took it out in a function in utils because it was used in many places and I was tired of duplicating it, it's exactly the same as before except that the code is in a different place